File: src/dataset/javascript/raindrops_exercism-javascript-programmiri/openAI/ChatGPT4_raindrops_v4.js
export const convert = (input) => {
	const divisibleBy = (n) => input % n === 0;

	if (!divisibleBy(3) && !divisibleBy(5) && !divisibleBy(7)) return input.toString();

	let result = '';
	if (divisibleBy(3)) result += 'Pling';
	if (divisibleBy(5)) result += 'Plang';
	if (divisibleBy(7)) result += 'Plong';

	return result;
};

File: src/dataset/c/raindrops_Exercism_attilahorvath/openAI/ChatGPT4_raindrops_v4.c
#include "raindrops.h"
#include <stdio.h>

const char *convert(char *buffer, size_t buffer_length, int number) {
    int div3 = (number % 3 == 0);
    int div5 = (number % 5 == 0);
    int div7 = (number % 7 == 0);

    if (div3 || div5 || div7) {
        char *p = buffer;
        size_t remaining = buffer_length;

        if (div3 && remaining > 6) {
            const char *s = "Pling";
            while (*s) { *p++ = *s++; remaining--; }
        }
        if (div5 && remaining > 6) {
            const char *s = "Plang";
            while (*s) { *p++ = *s++; remaining--; }
        }
        if (div7 && remaining > 6) {
            const char *s = "Plong";
            while (*s) { *p++ = *s++; remaining--; }
        }
        *p = '\0';
        return buffer;
    }

    snprintf(buffer, buffer_length, "%d", number);
    return buffer;
}
